### DISCUSSION TITLE
Confusion about cshtml files for Shortcodes

### DISCUSSION THREAD

**Discussion by devlife** [OrchardCMS/OrchardCore#11302] [Q&A] [answered]
I'm a bit confused about the contents of [this video](https://www.youtube.com/watch?v=ofPKGsW5Ftg&feature=emb_title)  about `IShortCodeProvider` implementations and the usage of cshtml files. Is this implemented? How would I go about creating a short code and having it utilize a cshtml file? Is the current approach basically to try and screenshot the code in the video? Not being sarcastic. I definitely get it if there is documentation missing. Just trying to figure out the current state of things.

**Comment by jtkech** [comment]
From memory when you define a custom short code, you can use liquid syntax, and then the liquid shape tag to render a shape ;)

**Reply by devlife** [reply]
I know you can do that but I have a scenario that would be great to have a cshtml file with which to render, make api calls, etc.

**Comment by jtkech** [comment]
You can create a `SomeName.cshtml` file and then target it through the shape tag helper

Something like `{% shape "SomeName", ... %}`

**Comment by Skrypt** [comment, marked as answer]
{% shape "SomeName", ... %}

Means you can basically use a SomeName.liquid or SomeName.cshtml template in your theme or admin UI templates to render some content. It's an "inline named shape" in which you can call any API's to render something.

**Reply by devlife** [reply]
I actually did try using that in the shortcode and couldn't get it to work. However, I will definitely give it a try again.
